contracts/voting/cwd-voting-cw20-staked/src/contract.rs

```diff
@@ -137,8 +138,27 @@ pub fn instantiate(
                 }
             }
 
+            // Has the acceptable side-effect of pre-sorting the vesting schedules chronologically
+            if let Some(vesting_info) = &mut vesting_info {
+                let mut vest_total = Uint128::from(0u128);
+                let schedules = &mut vesting_info.schedules;
+                for schedule in schedules.into_iter() {
+                    schedule.vests.sort_by_key(|v| v.expiration);
+                    if let Some(last) = schedule.vests.last() {
+                        vest_total = vest_total.checked_add(last.amount)?;
```

Comment:
isn't the total decided by the last point in the vesting schedule? why sum here?

Reply:
Each recipient has their own vesting schedule defined in absolute token terms. So we need to sum up the endpoints to get the total that will be vested to all recipients.